src/prefect/utilities/importtools.py

```diff
@@ -112,15 +115,9 @@ def load_script_as_module(path: str) -> ModuleType:
         with _get_sys_path_lock():
             sys.path.insert(0, working_directory)
             sys.path.insert(0, parent_path)
-            try:
-                spec.loader.exec_module(module)
-            finally:
-                sys.path.remove(parent_path)
-                sys.path.remove(working_directory)
+            spec.loader.exec_module(module)
     except Exception as exc:
         raise ScriptError(user_exc=exc, path=path) from exc
-    finally:
-        sys.modules.pop(module_name)
```

Comment:
I believe that all of this cleanup was implemented as a matter of best practice to avoid mutation, but I think in this case we want the changes to persist for pickling purposes.